front/lib/swr/data_source_views.ts

```diff
@@ -160,6 +161,101 @@ export function useDataSourceViewContentNodes({
   };
 }
 
+export function useDataSourceViewContentNodesWithInfiniteScroll({
+  owner,
+  dataSourceView,
+  internalIds,
+  parentId,
+  pageSize = 50,
+  viewType,
+}: {
+  owner: LightWorkspaceType;
+  dataSourceView?: DataSourceViewType;
+  internalIds?: string[];
+  parentId?: string;
+  pageSize?: number;
+  viewType?: ContentNodesViewType;
+}): {
+  isNodesError: boolean;
+  isNodesLoading: boolean;
+  isNodesValidating: boolean;
+  mutateDataSourceViewContentNodes: KeyedMutator<GetDataSourceViewContentNodes>;
+  nodes: GetDataSourceViewContentNodes["nodes"];
+  totalNodesCount: number;
+  hasMore: boolean;
+  nextPage: () => Promise<void>;
+} {
+  const url =
+    dataSourceView && viewType
+      ? `/api/w/${owner.sId}/vaults/${dataSourceView.vaultId}/data_source_views/${dataSourceView.sId}/content-nodes`
+      : null;
+
+  const body = {
+    internalIds,
+    parentId,
+    viewType,
+  };
+
+  const fetcher: Fetcher<GetDataSourceViewContentNodes, [string, object]> =
+    postFetcher;
+
+  const { data, error, setSize, size, isValidating, mutate } =
+    useSWRInfiniteWithDefaults(
+      (index) => {
+        if (!url) {
+          // No URL, return an empty array to skip the fetch
+          return "[]";
+        }
+
+        // Append the pagination params to the URL
+        const params = new URLSearchParams();
+        appendPaginationParams(params, {
+          pageIndex: index,
+          pageSize,
+        });
+
+        return JSON.stringify([url + "?" + params.toString(), body]);
+      },
+      async (fetchKey) => {
+        // Get the URL and body from the fetchKey
+        const params = JSON.parse(fetchKey);
+        if (!params[0]) {
+          // Empty array, skip the fetch
+          return undefined;
+        }
+
+        return fetcher(params);
+      },
+      {
+        revalidateFirstPage: false,
+      }
+    );
+
+  const mutateDataSourceViewContentNodes = async () => {
```

Comment:
What is this used for?

Reply:
The plan (discussed IRL with @Duncid  and @fontanierh ) is to use the infinite scroll also in tables, and mutate is used in VaultDatsSourceViewContentList . We need to redefine in order to flatten the results. Agree it's not used for now in trees.

Reply:
Not sure to follow but if code is not used, then code should not be here 😛 ?

Reply:
ok ok will reintroduce later :-)